### PR TITLE
Completely rewrite code to use Path instead of File

### PR DESCRIPTION
**Pull request by AlexProgrammerDE**

### CODE REVIEW

**Comment by AlexProgrammerDE** [comment]
Also rewrites default config values. Will continue work when I have some more time.

File: shared/src/main/java/net/skinsrestorer/shared/storage/MySQL.java [hunk 1]
```diff
@@ -58,66 +58,25 @@ public void createTable() {
...
-        new org.mariadb.jdbc.Driver();
-
-        con = DriverManager.getConnection("jdbc:mysql://" + host + ":" + port + "/" + database + "?" + options, username, password);
+    public void connectPool() throws SQLException {
+        poolDataSource = new MariaDbPoolDataSource(
+                "jdbc:mysql://" + host + ":" + port + "/" + database +
+                        "?user=" + username +
+                        "&password=" + password +
+                        "&permitMysqlScheme" +
+                        "&maxPoolSize=10" +
```
Comment (xknat) [inline shared/src/main/java/net/skinsrestorer/shared/storage/MySQL.java]: move to config mysql.options

File: shared/src/main/java/net/skinsrestorer/shared/storage/SkinStorage.java [hunk 2]
```diff
@@ -399,19 +403,22 @@ public void setSkinData(String skinName, IProperty textures, long timestamp) {
...
             }
         } else { // When not using mysql
-            // Filter out non "*.skin" files.
-            FilenameFilter skinFileFilter = (dir, name) -> name.endsWith(".skin");
+            List<Path> files = new ArrayList<>();
+            try (DirectoryStream<Path> stream = Files.newDirectoryStream(skinsFolder, "*.skin")) {
+                for (Path file : stream) {
+                    files.add(file);
+                }
+            } catch (IOException e) {
```
Comment (xknat) [inline shared/src/main/java/net/skinsrestorer/shared/storage/SkinStorage.java]: Don't you want to make this stream?

File: shared/src/main/java/net/skinsrestorer/shared/storage/SkinStorage.java [hunk 3]
```diff
@@ -455,27 +462,29 @@ public void setSkinData(String skinName, IProperty textures, long timestamp) {
...
-            // filter out non "*.skin" files.
-            FilenameFilter skinFileFilter = (dir, name) -> name.endsWith(".skin");
+            List<Path> files = new ArrayList<>();
 
-            String[] fileNames = skinsFolder.list(skinFileFilter);
+            try (DirectoryStream<Path> stream = Files.newDirectoryStream(skinsFolder, "*.skin")) {
+                for (Path file : stream) {
+                    files.add(file);
+                }
+            } catch (IOException e) {
```
Comment (xknat) [inline shared/src/main/java/net/skinsrestorer/shared/storage/SkinStorage.java]: Don't you want to make this stream?

**Review by xknat — CHANGES_REQUESTED** [review]

**Review by xknat — APPROVED** [review]
lgtm

**Pull request merged**